### ISSUE TITLE
systemd-executor should not call log_set_always_reopen_console(true) when running tests

### ISSUE THREAD

**Issue by daandemeyer** [systemd/systemd#32414]
### systemd version the issue has been seen with

main

### CPU architectures issue was seen on

None

### Unexpected behaviour you saw

`log_set_always_reopen_console(true)` means we will never log to stderr, which means that either we get the systemd-executor logs on /dev/console, or we got no logs at all during tests such as `test-execute` when running it from a terminal.

**Comment by bluca** [comment]
there were some early boot/late shutdown shenanigans there, that I cannot quite remember the details of

**Comment by DaanDeMeyer** [comment]
I'm not saying we should drop it unconditionally, but when running unit tests it does not make sense to enable it.

**Comment by bluca** [comment]
Yeah that makes sense, also dropping it if it doesn't cause any issues or there are other workarounds would be fine
